homeassistant/components/overkiz/water_heater_entities/domestic_hot_water_production.py

```diff
@@ -0,0 +1,281 @@
+"""Support for DomesticHotWaterProduction."""
+from __future__ import annotations
+
+from typing import Any, cast
+
+from pyoverkiz.enums import OverkizCommand, OverkizCommandParam, OverkizState
+
+from homeassistant.components.water_heater import (
+    STATE_ECO,
+    STATE_HIGH_DEMAND,
+    STATE_OFF,
+    STATE_PERFORMANCE,
+    WaterHeaterEntity,
+    WaterHeaterEntityFeature,
+)
+from homeassistant.const import ATTR_TEMPERATURE, TEMP_CELSIUS
+
+from ..entity import OverkizEntity
+
+OVERKIZ_TO_OPERATION_MODE: dict[str, str] = {
+    OverkizCommandParam.STANDARD: STATE_ECO,
+    OverkizCommandParam.HIGH_DEMAND: STATE_HIGH_DEMAND,
+    OverkizCommandParam.STOP: STATE_OFF,
+    OverkizCommandParam.MANUAL_ECO_ACTIVE: STATE_ECO,
+    OverkizCommandParam.MANUAL_ECO_INACTIVE: STATE_OFF,
+    OverkizCommandParam.AUTO: STATE_ECO,
+    OverkizCommandParam.AUTO_MODE: STATE_ECO,
```

Comment:
I think you can add `STATE_AUTO` to the main `homeassistant.components.water_heater` component, it won't be an issue on the core repo, seeing the usage of the consts.

But could be a `frontend` issue, so we should wheck that first.

Only places I see usage of `water_heater` constants (byGitHub):
- mostly `econet` : https://github.com/home-assistant/core/blob/fff967ce17e7b8360d136537367eb530928345ec/homeassistant/components/econet/water_heater.py#L28-L36
- `atag`: https://github.com/home-assistant/core/blob/fff967ce17e7b8360d136537367eb530928345ec/homeassistant/components/atag/water_heater.py#L16
- `hive`: https://github.com/home-assistant/core/blob/fff967ce17e7b8360d136537367eb530928345ec/homeassistant/components/hive/water_heater.py#L41

Reply:
@Quentame Not sure if this the right time for that.
See this #81913 and the corresponding discussion and all of it's replies [here](https://github.com/home-assistant/architecture/discussions/518#discussioncomment-4101948)

Let me know what you think.

Reply:
Yes sure !
That was an idea, but changing a base entity should be in a separate PR anyway, even more if there are architectural discussions about it 😅